File: launchdarkly/python/src/PatchTeamsExample.py
import json
from datetime import date, datetime
from pprint import pprint

from launchdarkly_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"ApiKey": "YOUR_API_KEY"},
)

with ApiClient(configuration) as api_client:
    teams_patch_input = models.TeamsPatchInput(
        instructions=json.loads("""
            [
                {
                    "kind": "addMembersToTeams",
                    "memberIDs": [
                        "1234a56b7c89d012345e678f"
                    ],
                    "teamKeys": [
                        "example-team-1",
                        "example-team-2"
                    ]
                }
            ]
        """),
        comment="Optional comment about the update",
    )

    try:
        response = api.TeamsBetaApi(api_client).patch_teams(
            teams_patch_input=teams_patch_input,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling TeamsBeta#patch_teams: %s\n" % e)

File: launchdarkly/ruby/src/PatchTeamsExample.rb
require "json"
require "launchdarkly_client"

LaunchDarklyClient.configure do |config|
    config.api_key["ApiKey"] = "YOUR_API_KEY"
end

teams_patch_input = LaunchDarklyClient::TeamsPatchInput.new
teams_patch_input.instructions = JSON.parse(<<-EOD
    [
        {
            "kind": "addMembersToTeams",
            "memberIDs": [
                "1234a56b7c89d012345e678f"
            ],
            "teamKeys": [
                "example-team-1",
                "example-team-2"
            ]
        }
    ]
    EOD
)
teams_patch_input.comment = "Optional comment about the update"

begin
    response = LaunchDarklyClient::TeamsBetaApi.new.patch_teams(
        teams_patch_input,
    )

    p response
rescue LaunchDarklyClient::ApiError => e
    puts "Exception when calling TeamsBetaApi#patch_teams: #{e}"
end
